lib/snippets/deploy-to-gke

```diff
@@ -0,0 +1,147 @@
+#!/usr/bin/env ruby
+
+# Usage: deploy-to-gke <app's namespace> <gcloud deployment> <gcloud project>
+
+# Requires the following to be available on Shipit machine:
+#  - gcloud binary available on the shipit machine's path
+#  - kubectl binary available in the shipit machine's path
+#  - $GCLOUD_CREDENTIALS_DIR/my-gcloud-project-name.json must exist
+
+# Optionally, the following variables can be used to override script defaults:
+#  - K8S_TEMPLATE_FOLDER: location of Kubernetes files to deploy. Default is config/<environment>/.
+
+require 'open3'
+require 'securerandom'
+require 'erb'
+require 'json'
+require 'yaml'
+require 'shellwords'
+require 'tempfile'
+
+class GKEDeployment
+  class FatalDeploymentError < StandardError; end
+
+  def initialize(namespace:, gcloud_deployment:, gcloud_project:, environment:, current_sha:, template_folder: nil)
+    @namespace = namespace
+    @gcloud_deployment = gcloud_deployment
+    @gcloud_project = gcloud_project
+    @current_sha = current_sha
+    enforce_required_parmas
+    @id = current_sha + "-#{SecureRandom.hex(4)}"
+    @template_path = './' + (template_folder || "config/deploy/#{environment}")
+
+    raise FatalDeploymentError, "Project config missing at #{ENV['GCLOUD_CREDENTIALS_DIR']}/#{@gcloud_project}.json" unless File.file?("#{ENV['GCLOUD_CREDENTIALS_DIR']}/#{@gcloud_project}.json")
```

Comment:
Nit: `#{ENV['GCLOUD_CREDENTIALS_DIR']}/#{@gcloud_project}.json` is in here three times. Should we make an `@key_file` or something?